source/shared/core_stmt.cpp

```diff
@@ -1266,11 +1265,17 @@ bool core_sqlsrv_send_stream_packet( sqlsrv_stmt* stmt TSRMLS_DC )
 
                 // the size of wbuffer is set for the worst case of UTF-8 to UTF-16 conversion, which is a 
                 // expansion of 2x the UTF-8 size.
-                wchar_t wbuffer[ PHP_STREAM_BUFFER_SIZE + 1 ];
-                // buffer_size is the # of wchars.  Since it set to stmt->param_buffer_size / 2, this is accurate
-                int wsize = MultiByteToWideChar( stmt->current_stream.encoding, MB_ERR_INVALID_CHARS, 
-                                                 buffer, static_cast<int>( read ), wbuffer, static_cast<int>( sizeof( wbuffer ) / sizeof( wchar_t )));
-                if( wsize == 0 && GetLastError() == ERROR_NO_UNICODE_TRANSLATION ) {
+                SQLWCHAR wbuffer[ PHP_STREAM_BUFFER_SIZE + 1 ];
+				DWORD last_error_code = ERROR_SUCCESS;
+				// buffer_size is the # of wchars.  Since it set to stmt->param_buffer_size / 2, this is accurate
+#ifndef _WIN32
+                int wsize = SystemLocale::ToUtf16Strict( stmt->current_stream.encoding, buffer, static_cast<int>(read), wbuffer, static_cast<int>(sizeof( wbuffer ) / sizeof( SQLWCHAR )), &last_error_code );
+#else
+                int wsize = MultiByteToWideChar( stmt->current_stream.encoding, MB_ERR_INVALID_CHARS, buffer, static_cast<int>( read ), wbuffer, static_cast<int>( sizeof( wbuffer ) / sizeof( wchar_t )));
```

Comment:
For consistency, should we use ```SQLWCHAR``` rather than ```wchar_t```?
If so, we could write a single line for the division of ```sizeof( wbuffer ) / sizeof( SQLWCHAR ))
Besides, should we consider floor() or ceil() before casting the result to an integer?